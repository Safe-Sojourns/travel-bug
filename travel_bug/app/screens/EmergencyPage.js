import React from 'react';
import {
  Button,
  View,
  Text,
  StyleSheet,
  SafeAreaView,
  TextInput,
  Image,
  ImageBackground,
} from 'react-native';
import {FontAwesomeIcon} from '@fortawesome/react-native-fontawesome';
import {
  faAmbulance,
  faUser,
  faInfoCircle,
} from '@fortawesome/free-solid-svg-icons';
import Modal from 'react-native-modal';

function EmergencyPage(props) {
  const [emergencyInput, onChangeText] = React.useState('');
  const [modalVisible, setModalVisible] = React.useState(false);

  return (
    <SafeAreaView style={styles.container}>
       {/* <View>
        <Image
          style={{
            height: 800,
            width: 400,
            resizeMode: 'stretch',
            // position: 'absolute',
            // top: 0,
            // left: 0,
            opacity: 0.06,
          }}
          source={require('./assets/travelbackground.jpeg')}
        />
      </View> */}
      <View>
        <View style={styles.card}>
          <View style={styles.iconText}>
            <FontAwesomeIcon
              style={styles.icon}
              icon={faAmbulance}
              size={25}
              accessibilityLabel="Ambulance"
            />
            <Text style={styles.text}>Emergency:</Text>
          </View>
          <View style={styles.iconText}>
            <FontAwesomeIcon
              style={styles.icon}
              icon={faUser}
              size={25}
              accessibilityLabel="User"
            />
            <Text style={styles.text}>Staff:</Text>
            <Text style={styles.textNum}>425-567-8976</Text>
          </View>
          <View style={styles.iconText}>
            <FontAwesomeIcon
              style={styles.icon}
              icon={faInfoCircle}
              size={25}
              accessibilityLabel="Info"
            />
            <Text style={styles.text}>Additional Information:</Text>
          </View>
          <View style={styles.info}>
            <Text style={styles.infoText}>{emergencyInput}</Text>
          </View>
          <View style={styles.buttonView}>
            <Button
              color="white"
              title="Edit"
              onPress={() => setModalVisible(true)}
            />
          </View>
          <View>
            <Modal isVisible={modalVisible}>
              <View style={styles.modal}>
                <View containerStyle={styles.cardModal} title="title">
                  <TextInput
                    style={styles.input}
                    value={emergencyInput}
                    onChangeText={text => onChangeText(text)}
                  />
                  <View style={styles.buttonView}>
                    <Button
                      color="white"
                      title="Save"
                      onPress={() => setModalVisible(false)}
                    />
                  </View>
                </View>
              </View>
            </Modal>
          </View>
        </View>
      </View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    // backgroundColor: '#EAF9FF',
    alignItems: 'center',
    justifyContent: 'center',
    // position: 'relative',
    // left: 30,
    // bottom: 200,
  },
  modal: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  buttonView: {
    width: 70,
    height: 40,
    backgroundColor: '#013220',
    alignSelf: 'flex-end',
    marginTop: 10,
    marginBottom: 5,
    borderRadius: 5,
    marginRight: 12,
  },
  card: {
    backgroundColor: '#ABDA9A',
    width: 300,
    height: 350,
    padding: 20,
    borderRadius: 15,
    // opacity: 1,
  },
  cardModal: {
    backgroundColor: '#ABDA9A',
    width: 300,
    height: 250,
    padding: 20,
    borderRadius: 5,
  },
  iconText: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  input: {
    height: 120,
    width: 250,
    backgroundColor: 'white',
    borderWidth: 1,
    paddingTop: 10,
  },
  info: {
    height: 120,
    width: 250,
    backgroundColor: '#EAF9FF',
    borderWidth: 1,
    borderRadius: 5,
    marginTop: 10,
  },
  text: {
    paddingTop: 10,
    paddingBottom: 10,
    paddingLeft: 10,
    color: '#013220',
    fontSize: 20,
    fontWeight: 'bold',
  },
  titleText: {
    color: '#013220',
    fontSize: 20,
  },
  textNum: {
    paddingLeft: 5,
    color: '#013220',
    fontSize: 20,
  },
  infoText: {
    fontSize: 16,
    color: '#013220',
  },
  icon: {
    color: '#013220',
  },
});

export default EmergencyPage;
