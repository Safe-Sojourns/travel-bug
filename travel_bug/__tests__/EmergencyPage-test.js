import React from 'react';
import renderer from 'react-test-renderer';
import EmergencyPage from '../app/screens/EmergencyPage.js';

jest.useFakeTimers();

jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));

const tree = renderer.create(<EmergencyPage />);

test('Emergency renders correctly', () => {
  expect(tree).toMatchSnapshot();
});

test('Modal appears', () => {
  const button = tree.root.findByProps({testID: 'editButton'}).props;
  renderer.act(() => button.onPress());

  const modal = tree.root.findByProps({testID: 'modal'}).props;
  expect(modal).toBeDefined();
});

test('Save button appears', () => {
  const button = tree.root.findByProps({testID: 'editButton'}).props;
  renderer.act(() => button.onPress());

  const save = tree.root.findByProps({testID: 'saveButton'}).props;
  expect(save).toBeDefined();
});

test('Input appears', () => {
  const button = tree.root.findByProps({testID: 'editButton'}).props;
  renderer.act(() => button.onPress());

  const input = tree.root.findByProps({testID: 'input'}).props;
  expect(input).toBeDefined();
});

test('View appears', () => {
  const button = tree.root.findByProps({testID: 'editButton'}).props;
  renderer.act(() => button.onPress());

  const view = tree.root.findByProps({testID: 'view'}).props;
  expect(view).toBeDefined();
});

test('Emergency number exists', () => {
  const emergency = tree.root.findByProps({testID: 'emergency'}).props;
  expect(emergency).toBeDefined();
});

test('Staff number exists', () => {
  const staff = tree.root.findByProps({testID: 'staff'}).props;
  expect(staff).toBeDefined();
});

test('Additional info exists', () => {
  const addInfo = tree.root.findByProps({testID: 'addInfo'}).props;
  expect(addInfo).toBeDefined();
});
