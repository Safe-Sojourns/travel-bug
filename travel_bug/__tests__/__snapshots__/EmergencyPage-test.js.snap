// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#EAF9FF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ABDA9A",
        "borderRadius": 15,
        "height": 350,
        "padding": 20,
        "width": 300,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <
        accessibilityLabel="Ambulance"
        icon={
          Object {
            "icon": Array [
              640,
              512,
              Array [],
              "f0f9",
              "M624 352h-16V243.9c0-12.7-5.1-24.9-14.1-33.9L494 110.1c-9-9-21.2-14.1-33.9-14.1H416V48c0-26.5-21.5-48-48-48H48C21.5 0 0 21.5 0 48v320c0 26.5 21.5 48 48 48h16c0 53 43 96 96 96s96-43 96-96h128c0 53 43 96 96 96s96-43 96-96h48c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zM160 464c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48zm144-248c0 4.4-3.6 8-8 8h-56v56c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8v-56h-56c-4.4 0-8-3.6-8-8v-48c0-4.4 3.6-8 8-8h56v-56c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v56h56c4.4 0 8 3.6 8 8v48zm176 248c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48zm80-208H416V144h44.1l99.9 99.9V256z",
            ],
            "iconName": "ambulance",
            "prefix": "fas",
          }
        }
        size={25}
        style={
          Object {
            "color": "#013220",
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#013220",
            "fontSize": 20,
            "fontWeight": "bold",
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingTop": 10,
          }
        }
      >
        Emergency:
      </Text>
      <Text
        style={
          Object {
            "color": "#013220",
            "fontSize": 20,
            "paddingLeft": 5,
          }
        }
      >
        911
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <
        accessibilityLabel="User"
        icon={
          Object {
            "icon": Array [
              448,
              512,
              Array [],
              "f007",
              "M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z",
            ],
            "iconName": "user",
            "prefix": "fas",
          }
        }
        size={25}
        style={
          Object {
            "color": "#013220",
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#013220",
            "fontSize": 20,
            "fontWeight": "bold",
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingTop": 10,
          }
        }
      >
        Staff:
      </Text>
      <Text
        style={
          Object {
            "color": "#013220",
            "fontSize": 20,
            "paddingLeft": 5,
          }
        }
      >
        425-567-8976
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <
        accessibilityLabel="Info"
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f05a",
              "M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z",
            ],
            "iconName": "info-circle",
            "prefix": "fas",
          }
        }
        size={25}
        style={
          Object {
            "color": "#013220",
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#013220",
            "fontSize": 20,
            "fontWeight": "bold",
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingTop": 10,
          }
        }
      >
        Additional Information:
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#EAF9FF",
          "borderRadius": 5,
          "borderWidth": 1,
          "height": 120,
          "marginTop": 10,
          "width": 250,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#013220",
            "fontSize": 16,
          }
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "flex-end",
          "backgroundColor": "#013220",
          "borderRadius": 5,
          "height": 40,
          "marginBottom": 5,
          "marginRight": 12,
          "marginTop": 10,
          "width": 70,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "white",
                },
              ]
            }
          >
            Edit
          </Text>
        </View>
      </View>
    </View>
    <View>
      <Modal
        animationType="none"
        deviceHeight={null}
        deviceWidth={null}
        hardwareAccelerated={false}
        hideModalContentWhileAnimating={false}
        onBackdropPress={[Function]}
        onModalHide={[Function]}
        onModalWillHide={[Function]}
        onModalWillShow={[Function]}
        onRequestClose={[Function]}
        scrollHorizontal={false}
        scrollOffset={0}
        scrollOffsetMax={0}
        scrollTo={null}
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        swipeThreshold={100}
        transparent={true}
        visible={false}
      >
        <View
          accessible={true}
          focusable={true}
          forwardedRef={[Function]}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "black",
              "bottom": 0,
              "height": 1334,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 750,
            }
          }
        />
        <View
          deviceHeight={null}
          deviceWidth={null}
          forwardedRef={[Function]}
          hideModalContentWhileAnimating={false}
          onBackdropPress={[Function]}
          onModalHide={[Function]}
          onModalWillHide={[Function]}
          onModalWillShow={[Function]}
          pointerEvents="box-none"
          scrollHorizontal={false}
          scrollOffset={0}
          scrollOffsetMax={0}
          scrollTo={null}
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
              "margin": 37.5,
              "transform": Array [
                Object {
                  "translateY": 0,
                },
              ],
            }
          }
          supportedOrientations={
            Array [
              "portrait",
              "landscape",
            ]
          }
          swipeThreshold={100}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              containerStyle={
                Object {
                  "backgroundColor": "#ABDA9A",
                  "borderRadius": 5,
                  "height": 250,
                  "padding": 20,
                  "width": 300,
                }
              }
              title="title"
            >
              <TextInput
                allowFontScaling={true}
                onChangeText={[Function]}
                rejectResponderTermination={true}
                style={
                  Object {
                    "backgroundColor": "white",
                    "borderWidth": 1,
                    "height": 120,
                    "paddingTop": 10,
                    "width": 250,
                  }
                }
                underlineColorAndroid="transparent"
                value=""
              />
              <View
                style={
                  Object {
                    "alignSelf": "flex-end",
                    "backgroundColor": "#013220",
                    "borderRadius": 5,
                    "height": 40,
                    "marginBottom": 5,
                    "marginRight": 12,
                    "marginTop": 10,
                    "width": 70,
                  }
                }
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={Object {}}
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {},
                      ]
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#007AFF",
                            "fontSize": 18,
                            "margin": 8,
                            "textAlign": "center",
                          },
                          Object {
                            "color": "white",
                          },
                        ]
                      }
                    >
                      Save
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </Modal>
    </View>
  </View>
</RCTSafeAreaView>
`;
